### PR TITLE
The Haag Beyond the Pines

### PR DESCRIPTION
**Pull request by MHaggis**
## Updated Analytic Story
- Malicious PowerShell

## New Analytics
- Detect Empire with PowerShell Script Block Logging (Passed)
- Detect Mimikatz With PowerShell Script Block Logging (Passed)
- Powershell Fileless Process Injection via GetProcAddress (Passed)
- Powershell Fileless Script Contains Base64 Encoded Content (Passed)
- Unloading AMSI via Reflection (Passed)
- PowerShell Domain Enumeration (Passed)
- PowerShell Loading DotNET into Memory via System Reflection Assembly (Passed)
- Detect WMI Event Subscription Persistence (Passed)

## Updated Analytics
- Remote WMI Command Attempt (Passed)
- Process Execution via WMI
- WMI Permanent Event Subscription - Sysmon

### CODE REVIEW

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect Empire with PowerShell Script Block Logging :white_check_mark:
- diskUsage: 167936
- runDuration: 0.795s
- scanCount: 17

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect Mimikatz With PowerShell Script Block Logging :white_check_mark:
- diskUsage: 167936
- runDuration: 1.059s
- scanCount: 228

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Unloading AMSI via Reflection :white_check_mark:
- diskUsage: 167936
- runDuration: 0.778s
- scanCount: 6

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Powershell Fileless Process Injection via GetProcAddress :white_check_mark:
- diskUsage: 163840
- runDuration: 1.22s
- scanCount: 228

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Powershell Fileless Script Contains Base64 Encoded Content :white_check_mark:
- diskUsage: 163840
- runDuration: 1.09s
- scanCount: 228

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

PowerShell Loading DotNET into Memory via System Reflection Assembly :white_check_mark:
- diskUsage: 167936
- runDuration: 0.903s
- scanCount: 8

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

PowerShell Domain Enumeration :white_check_mark:
- diskUsage: 172032
- runDuration: 1.082s
- scanCount: 47

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect WMI Event Subscription Persistence :x:
- diskUsage: 135168
- runDuration: 1.119s
- scanCount: 20

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect WMI Event Subscription Persistence :x:
- diskUsage: 135168
- runDuration: 0.976s
- scanCount: 59

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect WMI Event Subscription Persistence :x:
- diskUsage: 135168
- runDuration: 0.916s
- scanCount: 39

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect WMI Event Subscription Persistence :x:
- diskUsage: 131072
- runDuration: 1.116s
- scanCount: 39

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect WMI Event Subscription Persistence :x:
- diskUsage: 135168
- runDuration: 0.897s
- scanCount: 2

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect WMI Event Subscription Persistence :white_check_mark:
- diskUsage: 143360
- runDuration: 1.088s
- scanCount: 60

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Remote WMI Command Attempt :x:
- diskUsage: 409600
- runDuration: 2.008s
- scanCount: 1032

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Remote WMI Command Attempt :white_check_mark:
- diskUsage: 413696
- runDuration: 4.798s
- scanCount: 6315

**Review by patel-bhavin — APPROVED** [review]

File: detections/endpoint/detect_mimikatz_with_powershell_script_block_logging.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  \
+
+  This analytic identifies common Mimikatz functions that may be identified in the
+  script block, including `mimikatz`. This will catch the most basic use cases for
+  Pass the Ticket, Pass the Hash and `-DumprCreds`. \
+
+  During triage, review parallel processes using an EDR product or 4688 events. It
+  will be important to understand the timeline of events around this activity. Review
+  the entire logged PowerShell script block.'
+search: '`powershell` EventCode=4104 Message IN (*mimikatz*, *-dumpcr*, *sekurlsa::pth*,
```
Comment (miskoSplunk) [inline detections/endpoint/detect_mimikatz_with_powershell_script_block_logging.yml:21]: @MHaggis I did some of this for SSA - some of the things we could add here are common calls when someone just breaks in: 
- token::whoami
- privilege::debug (trying to become an admin)
- crypto::capi
- kerberos::list
- crypto::keys

File: detections/endpoint/detect_wmi_event_subscription_persistence.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  1. Consumer - An action to take upon triggering the filter. EventID = 20 \
+
+  1. Binding - Registers a filter to a consumer. EventID = 21 \
+
+  Monitor for the creation of new WMI EventFilter, EventConsumer, and FilterToConsumerBinding.
+  It may be pertinent to review all 3 to identify the flow of execution. In addition,
+  EventCode 4104 may assist with any other PowerShell script usage that registered
+  the subscription.'
+search: '`sysmon` EventID=20 | stats count min(_time) as firstTime max(_time) as lastTime
```
Comment (miskoSplunk) [inline detections/endpoint/detect_wmi_event_subscription_persistence.yml:28]: @MHaggis, are we sure that EventID=20 will only spot WMI?

File: detections/endpoint/wmi_permanent_event_subscription___sysmon.yml [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
+
+  1. Filter - WQL Query for the events we want. EventID = 19 \
+
+  1. Consumer - An action to take upon triggering the filter. EventID = 20 \
+
+  1. Binding - Registers a filter to a consumer. EventID = 21 \
+
+  Monitor for the creation of new WMI EventFilter, EventConsumer, and FilterToConsumerBinding. It may be pertinent to review all 3 to identify the flow of execution. In addition, EventCode 4104 may assist with any other PowerShell script usage that registered the subscription.
+
 search: '`sysmon` EventCode=21 | rename host as dest | table _time, dest, user, Operation,
```
Comment (miskoSplunk) [inline detections/endpoint/wmi_permanent_event_subscription___sysmon.yml:21]: ... also here - are we catching only WMi with event 21?

**Pull request merged**